src/main/java/com/minecolonies/coremod/compatibility/jei/LootTableAnalyzer.java

```diff
@@ -0,0 +1,198 @@
+                    packs.add(new ModFileResourcePack(mod.getFile()));
+                }
+                packs.forEach(serverResourceManger::addResourcePack);
+                serverResourceManger.addReloadListener(manager);
+
+                final CompletableFuture<Unit> completableFuture = serverResourceManger.reloadResourcesAndThen(Util.getServerExecutor(), Minecraft.getInstance(), packs, CompletableFuture.completedFuture(Unit.INSTANCE));
+                Minecraft.getInstance().driveUntil(completableFuture::isDone);
+            }
+            return manager;
+        }
+        return world.getServer().getLootTableManager();
+    }
+
+    public static JsonObject getLootTableJson(@NotNull final LootTable table)
+    {
+        return LootTableManager.toJson(table).getAsJsonObject();
+    }
+
+    public static JsonObject getLootTableJson(@NotNull final ResourceLocation tableId)
+    {
+        final LootTable table = getManager(null).getLootTableFromLocation(tableId);
+        return getLootTableJson(table);
+    }
+
+    public static List<LootDrop> toDrops(@NotNull final JsonObject lootTableJson)
+    {
+        final List<LootDrop> drops = new ArrayList<>();
+
+        final JsonArray pools = JSONUtils.getJsonArray(lootTableJson, "pools");
+        for (final JsonElement pool : pools)
+        {
+            final JsonArray entries = JSONUtils.getJsonArray(pool.getAsJsonObject(), "entries", new JsonArray());
+            final float totalWeight = StreamSupport.stream(entries.spliterator(), false)
+                    .filter(entry -> {
+                        final String type = JSONUtils.getString(entry.getAsJsonObject(), "type");
+                        return type.equals("minecraft:empty") || type.equals("minecraft:item") || type.equals("minecraft:tag") || type.equals("minecraft:loot_table");
+                    })
+                    .mapToInt(entry -> JSONUtils.getInt(entry.getAsJsonObject(), "weight", 1))
+                    .sum();
+
+            for (final JsonElement ej : entries)
+            {
+                final JsonObject entryJson = ej.getAsJsonObject();
+                final String type = JSONUtils.getString(entryJson, "type");
+                if (type.equals("minecraft:item"))
+                {
+                    final Item item = ForgeRegistries.ITEMS.getValue(new ResourceLocation(JSONUtils.getString(entryJson, "name")));
+                    final float weight = JSONUtils.getFloat(entryJson, "weight", 1);
+                    final boolean variableQuality = JSONUtils.getFloat(entryJson, "quality", 0) != 0;
+
+                    drops.add(new LootDrop(new ItemStack(item), weight / totalWeight, variableQuality));
+                }
+                else if (type.equals("minecraft:loot_table"))
+                {
+                    final ResourceLocation table = new ResourceLocation(JSONUtils.getString(entryJson, "name"));
+                    drops.addAll(toDrops(getLootTableJson(table)));

```

Comment:
You're not taking weight into account here, nor modifying the returns appropriately. The nether gatherer I'm building which is based on the current sifter code calls through to a wide variety of sub loot tables, and this will produce a very bad set of probabilities. 

Frankly, I'm not comfortable with attempting to recreate the loot table logic for display, for reasons of performance and complexity, especially client side. I'd rather we calculate the drops on the server and include them in the recipes across the wire.

Reply:
Well, currently there aren't actually any examples of a sub-loot-table.  So that could be looked into when that actually happens.

As for calculating it more directly on the server side, the main problem with that is that loot tables are not actually iterable, so you can't really get any answer better than "it might sometimes drop this item but it might be different in some way we didn't calculate" or by running a sample of say 1000 rolls on the loot table and estimating the results.

This particular code is not comprehensive but it does suffice for all currently existing loot tables.